File: programmers/level-1/12947. 하샤드 수/solution.cc
#include <string>
#include <vector>

using namespace std;

bool solution(int x) {
    int sum = 0, r = x;
    while (r > 0) {
        sum += r % 10;
        r = (int)(r / 10);
    }
    return x % sum == 0;
}

File: programmers/level-1/12947. 하샤드 수/solution.js
/**
 * https://programmers.co.kr/learn/courses/30/lessons/12947
 * @param {number} x 
 * @returns {boolean}
 */
const solution = x => {
    let s = 0, k = x;
    for (let i = 0; k > 0; i++) {
        s += k % 10;
        k = Math.floor(k / 10);
    }
    return x % s === 0;
};